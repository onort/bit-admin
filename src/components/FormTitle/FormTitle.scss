@import "../../styles/main.scss";

.container {
  display: flex;
  border-bottom: 1px solid $light-border;
  padding-bottom: $spacer-xxsm;
}

.icon {
  display: flex;
  align-items: center;
  font-size: $font-size-xxlg;
  margin-right: $spacer-sm;
}

.title {
  font-size: $font-size-xlg;
  font-weight: 600;
}
