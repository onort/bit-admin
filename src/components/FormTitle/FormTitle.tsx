import React from "react"
import cx from "classnames"

import styles from "./FormTitle.scss"

interface Props {
  className?: string
  icon?: React.ReactNode
  title: string
}

const FormTitle: React.SFC<Props> = props => {
  const className = cx(styles.container, props.className)
  return (
    <div className={className}>
      {props.icon && <span className={styles.icon}>{props.icon}</span>}
      <span className={styles.title}>{props.title}</span>
    </div>
  )
}

export default FormTitle
