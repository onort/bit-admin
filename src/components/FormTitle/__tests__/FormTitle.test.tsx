import React from "react"
import { shallow } from "enzyme"
import { MdAdd as AddIcon } from "react-icons/md"

import FormTitle from "../"

const testTitle = "Test"
const testClassName = "testClassName"

describe("<FormTitle />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<FormTitle icon={<AddIcon />} title={testTitle} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should render title", () => {
    const wrapper = shallow(<FormTitle title={testTitle} />)
    expect(wrapper.find(".title").text()).toBe(testTitle)
  })

  it("should render icon", () => {
    const wrapper = shallow(<FormTitle icon={<AddIcon />} title={testTitle} />)
    expect(wrapper.find(AddIcon).exists()).toBe(true)
  })

  it("should have custom classname", () => {
    const wrapper = shallow(
      <FormTitle className={testClassName} title={testTitle} />
    )
    expect(wrapper.find(".container").hasClass(testClassName)).toBe(true)
  })
})
