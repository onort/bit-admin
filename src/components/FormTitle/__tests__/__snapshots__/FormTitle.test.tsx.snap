// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTitle /> should match snapshot 1`] = `
<div
  className="container"
>
  <span
    className="icon"
  >
    <MdAdd />
  </span>
  <span
    className="title"
  >
    Test
  </span>
</div>
`;

exports[`<FormTitle /> should match snapshot 2`] = `"<div class=\\"container\\"><span class=\\"icon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></span><span class=\\"title\\">Test</span></div>"`;
